src/main/java/org/spongepowered/common/mixin/core/entity/MixinEntity.java

```diff
@@ -649,6 +652,14 @@ public boolean setLocationAndRotation(Location<World> location, Vector3d rotatio
         return relocated;
     }
 
+    private void checkWorldBorder(World world, Vector3d location) {
+        AxisAlignedBB aabb = this.getEntityBoundingBox().offset(location.getX() - this.posX, location.getY() - this.posY,
+                location.getZ() - this.posZ);
+        if (!((net.minecraft.world.World) world).worldBorder.contains(aabb)) {
```

Comment:
I really don't think this is what we want to do. It's perfectly fine for players and entities to be outside the WorldBorder - in fact, the [/wordborder command allows you to cause players to take damage when outside the border](https://minecraft.gamepedia.com/World_border).

Instead of checking `World.worldBorder`, you should check `ChunkLayout#toWorld`, to see if the coordinates actually correspond to a legal chunk (whether or not the x and z coordinates lie from `-30000000` to `30000000`)